cinder/volume/drivers/vmware/datastore.py

```diff
@@ -282,6 +288,7 @@ def select_datastore_by_name(self, name):
         resource_pool = self._get_resource_pool(parent)
         return (host, resource_pool, summary)
 
+    @cached(cache=TTLCache(maxsize=128, ttl=300))
```

Comment:
While `parent` should be stable and thus good for caching, `runtime` contains a lot of stuff that's not. We seem to use `runtime.connectionState` only afaics (I don't know why we don't just fetch that one key then, would probably be quicker). `connectionState` is quite important though, as we might try to use a host for worst-case 5min, that's already disconnected in the vCenter.